app/Repositories/Task/TaskRepository.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Repositories\Task;
+
+use App\Enums\StatusDelete;
+use App\Enums\TaskStatus;
+use App\Interfaces\Task\TaskRepositoryInterface;
+use App\Models\Task;
+use App\Repositories\BaseRepository;
+use Illuminate\Support\Facades\DB;
+
+class TaskRepository extends BaseRepository implements TaskRepositoryInterface
+{
+    public function __construct(Task $task)
+    {
+        $this->model = $task;
+    }
+
+    /**
+     * fetch all tasks with subtasks property
+     * @return \Illuminate\Database\Eloquent\Builder[]|\Illuminate\Database\Eloquent\Collection
+     */
+    public function getDataAllTaskWithSubtasks()
+    {
+        return $this->model->with('subtasks')
+            ->join('type_tasks', 'tasks.type_id', 'type_tasks.id')
```

Comment:
nữa nạc nữa mỡ zị em,
viết relationship,
$this->model->whereHas('typeTask')->with('subTasks')